metadata-io/src/test/java/com/linkedin/metadata/entity/DeleteReferencesServiceTest.java

```diff
@@ -0,0 +1,388 @@
+package com.linkedin.metadata.entity;
+
+import com.datahub.util.RecordUtils;
+import com.linkedin.data.DataList;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.schema.DataSchema;
+import com.linkedin.data.schema.PathSpec;
+import com.linkedin.data.schema.grammar.PdlSchemaParser;
+import com.linkedin.data.schema.resolver.DefaultDataSchemaResolver;
+import com.linkedin.entity.Aspect;
+import com.linkedin.schema.SchemaMetadata;
+import junit.framework.TestCase;
+import org.testng.annotations.Test;
+
+public class DeleteReferencesServiceTest extends TestCase {
+
+  /**
+   * Tests that Aspect Processor deletes the entire struct if it no longer has any fields
+   */
+  @Test
+  public void testEmptyStructRemoval() {
+    final String value = "{\"key_a\": \"hello\"}";
+    final Aspect aspect = RecordUtils.toRecordTemplate(Aspect.class, value);
+
+    final PdlSchemaParser pdlSchemaParser = new PdlSchemaParser(new DefaultDataSchemaResolver());
+    pdlSchemaParser.parse("record simple_record {\n"
+        + "key_a: optional string\n"
+        + "}");
+
+    final DataSchema schema = pdlSchemaParser.lookupName("simple_record");
+    final Aspect updatedAspect = DeleteReferencesService.getAspectWithReferenceRemoved("hello", aspect, schema,
+        new PathSpec("key_a"));
+
+    assertFalse(updatedAspect.data().containsKey("key_a"));
+    assertTrue(updatedAspect.data().isEmpty());
```

Comment:
Does this mean that if an aspect has a single field, the entire aspect will be removed? 

E.g. the "container" aspect, which has 1 "container" field

Reply:
I believe it does not, looking at rest of code

Reply:
If the entire aspect is empty it will be removed if it's a property of the aspect passed by input to the `getAspectWithReferenceRemoved` method, otherwise an empty aspect will be returned.

Reply:
Let's make sure the empty aspect case is well tested - we don't usually remove aspects by writing an empty one..

You could alternatively issue a "DELETE" operation on the aspect in question (inside an MCP)